src/h5web/visualizations/index.tsx

```diff
@@ -123,22 +120,14 @@ export const VIS_DEFS: Record<Vis, VisDef> = {
         return false; // entity is not a group or doesn't have a `default` attribute
       }
 
-      const signal = getAttributeValue(group, 'signal');
-      assertDefined(signal, "Expected 'signal' attribute");
-      assertStr(signal, "Expected 'signal' attribute to be a string");
-
-      const dataset = getLinkedEntity(signal, group, metadata);
-      assertDefined(dataset, `Expected "${signal}" signal entity to exist`);
-      assertDataset(dataset, `Expected "${signal}" signal to be a dataset`);
-      assertNumericType(dataset);
-      assertSimpleShape(dataset);
-
+      const signal = findSignalName(group);
+      const dataset = findSignalDataset(signal, group, metadata);
       const interpretation = getAttributeValue(dataset, 'interpretation');
-      if (isNxInterpretation(interpretation)) {
-        return interpretation === NxInterpretation.Spectrum;
-      }
 
-      return true;
+      return (
+        !isNxInterpretation(interpretation) ||
+        interpretation === NxInterpretation.Spectrum
+      );
```

Comment:
I beg to differ :sweat_smile: 

I find the previous writing much easier to read

Reply:
You don't think the `return true` on its own was confusing? Here we're saying, in one statement, that the vis supports the entity "if it doesn't have an interpretation or if the interpretation is `Spectrum`". We treat the generic negative case first, and we keep the most specific condition last.

Anyway, happy to revert if you're not convinced 😉.

Reply:
Written like this, it is clear but the problem is I don't read `it doesn't have an interpretation` when I see `!isNxInterpretation(interpretation)`...

This being said, the alternative `return true` is indeed confusing.

Let's keep your modification. I am going to assume that I had to read the line several times to understand it because I did not take my coffee yet :rofl:

Reply:
I could introduce a function called `hasNxInterpretation`:

```
return hasNxInterpretation(entity, NxInterpretation.Spectrum);
```

Reply:
The code inside the function would be exactly the same, but it'd be more readable here.

Reply:
Nah cause it means it will return `true` if the `entity` has no `NxInterpretation` which is weird considering the function name !

Reply:
Ah of course 🤦